src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs.rs

```diff
@@ -0,0 +1,229 @@
+//! This module provides a view for breadcrumbs, enabling us to know which node the graph being
+//! edited belongs to.
+
+use crate::prelude::*;
+
+pub mod breadcrumb;
+pub use breadcrumb::Breadcrumb;
+
+use enso_frp as frp;
+use ensogl::display;
+use ensogl::display::object::ObjectOps;
+use ensogl::display::scene::Scene;
+use logger::enabled::Logger;
+use logger::AnyLogger;
+
+
+
+// =================
+// === FrpInputs ===
+// =================
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct FrpInputs {
+    /// Push breadcrumb.
+    pub push_breadcrumb : frp::Source<String>,
+    /// Pop breadcrumb.
+    pub pop_breadcrumb : frp::Source,
+}
+
+impl FrpInputs {
+    /// Create new FrpInputs.
+    pub fn new(network:&frp::Network) -> Self {
+        frp::extend! {network
+            def push_breadcrumb = source();
+            def pop_breadcrumb  = source();
+        }
+        Self{push_breadcrumb,pop_breadcrumb}
+    }
+}
+
+
+
+// ==================
+// === FrpOutputs ===
+// ==================
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct FrpOutputs {
+    pub pushed_breadcrumb : frp::Source<String>,
+    pub popped_breadcrumb : frp::Source<String>
+}
+
+impl FrpOutputs {
+    /// Create new FrpOutputs.
+    pub fn new(network:&frp::Network) -> Self {
+        frp::extend! {network
+            def pushed_breadcrumb = source();
+            def popped_breadcrumb = source();
+        }
+        Self{pushed_breadcrumb,popped_breadcrumb}
+    }
+}
+
+
+
+// ===========
+// === Frp ===
+// ===========
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct Frp {
+    pub inputs  : FrpInputs,
+    pub outputs : FrpOutputs,
+    pub network : frp::Network,
+}
+
+impl Deref for Frp {
+    type Target = FrpInputs;
+    fn deref(&self) -> &Self::Target {
+        &self.inputs
+    }
+}
+
+impl Default for Frp {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Frp {
+    /// Create new Frp.
+    pub fn new() -> Self {
+        let network = frp::Network::new();
+        let inputs  = FrpInputs::new(&network);
+        let outputs = FrpOutputs::new(&network);
+        Self{network,inputs,outputs}
+    }
+}
+
+
+
+// ==================
+// === Animations ===
+// ==================
+
+/// ProjectName's animations handlers.
+#[derive(Debug,Clone,CloneRef,Copy)]
+pub struct Animations {}
+
+impl Animations {
+    /// Create new animations handlers.
+    pub fn new(_network:&frp::Network) -> Self {
+        Self{}
+    }
+}
```

Comment:
What woudl it be used for ? Animations are part of frp network, you do not need to handle the explicitly